resource_obmcs_identity_policy.go

```diff
@@ -165,23 +205,31 @@ func (s *PolicyResourceCrud) Update() (e error) {
 	}
 
 	if rawStatements, ok := s.D.GetOk("statements"); ok {
-		statements := s.toStringArray(rawStatements)
+		statements := toStringArray(rawStatements)
 		opts.Statements = statements
+		s.D.Set("policyHash", getMD5Hash(statements))
 	}
 
 	s.Res, e = s.Client.UpdatePolicy(s.D.Id(), opts)
+	s.D.Set("lastUpdateETag", s.Res.ETag)
 	return
 }
 
 func (s *PolicyResourceCrud) SetData() {
 	s.D.Set("statements", s.Res.Statements)
+	s.D.Set("ETag", s.Res.ETag)
 	s.D.Set("name", s.Res.Name)
 	s.D.Set("description", s.Res.Description)
 	s.D.Set("compartment_id", s.Res.CompartmentID)
 	s.D.Set("state", s.Res.State)
 	s.D.Set("time_created", s.Res.TimeCreated.String())
 }
 
+func getMD5Hash(values [] string) string {
+	statements := strings.Join(values, "#");
+	return fmt.Sprintf("%x", md5.Sum([]byte(statements)))
```

Comment:
Have a look at `hex.EncodeToString`

Reply:
hex.EncodeToString  encodes the string into hexadecimal format, I want hash so that we can do equality comparison without having to storE the entire policy statement.

 Are you suggesting to do hex encoding before hashing or do you want me to store encoded string?

Reply:
You are using fmt.Sprintf to coerce a hex value to a string, I was suggesting using the intended utility for that conversion

Reply:
Oh, Ok. Done